### PR TITLE
Fixes #1 basic nsx provider files

### LINKED ISSUE #1: Add vmware-nsx provoder basic code
main.go
Makefile
provider and config go files

### PR DESCRIPTION
**Pull request by rachappag**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+deps:
+	go get github.com/go-resty/resty
+	
+build: 
+	go build -o terraform-provider-nsx main.go
+
+fmt:
+	go fmt ./nsx 
+	go fmt ../govnsx
```
Comment (chandidas) [inline Makefile]: Do we need to fmt govnsx code from nsx-provider make?

File: nsx/config.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up client debug: %s", err)
+	}
+
+	nsxMrgParams := &govnsx.NsxManagerParams{
+		UserName:      c.User,
+		Password:      c.Password,
+		Uri:           c.NsxManagerUri,
+		AllowInsecssl: c.InsecureFlag,
+		RestAgentName: "govnsx",
```
Comment (chandidas) [inline nsx/config.go]: RestAgentName should be configurable with some default value "Terraform-Nsx-Provider"

File: nsx/config.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// Client() returns a new client for accessing NSX manager
+//
+func (c *Config) Client() (*govnsx.Client, error) {
+
+	err := c.EnableDebug()
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up client debug: %s", err)
+	}
+
+	nsxMrgParams := &govnsx.NsxManagerParams{
```
Comment (chandidas) [inline nsx/config.go]: change to nsxMgrParams

File: nsx/config.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	DebugPath     string
+	DebugPathRun  string
+}
+
+//
+// Client() returns a new client for accessing NSX manager
+//
+func (c *Config) Client() (*govnsx.Client, error) {
+
+	err := c.EnableDebug()
```
Comment (chandidas) [inline nsx/config.go:29]: Should not this be called after client object creation?

**Review by chandidas — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rachappag) [reply Makefile]: removed fmt exec for govnsx from nsx provider Make file

File: nsx/config.go (hunk 4 above)
Comment (rachappag) [reply nsx/config.go:29]: even before we create the client object create, we need to set the debug related facility to cover the client object creation code too

File: nsx/config.go (hunk 3 above)
Comment (rachappag) [reply nsx/config.go]: changed

File: nsx/config.go (hunk 2 above)
Comment (rachappag) [reply nsx/config.go]: changed

**Review by chandidas — APPROVED** [review]

**Pull request merged**